api/src/services/relations.ts

```diff
@@ -47,6 +50,30 @@ export class RelationsService {
 		this.helpers = getHelpers(this.knex);
 	}
 
+	async foreignKeys(collection?: string) {
+		const schemaCacheIsEnabled = Boolean(env['CACHE_SCHEMA']);
+
+		let foreignKeys: ForeignKey[] | null = null;
+
+		if (schemaCacheIsEnabled) {
+			foreignKeys = await getCacheValue(this.systemCache, 'foreignKeys');
+		}
+
+		if (!foreignKeys) {
+			foreignKeys = await this.schemaInspector.foreignKeys();
+
+			if (schemaCacheIsEnabled) {
+				setCacheValue(this.systemCache, 'foreignKeys', foreignKeys);
```

Comment:
Let's cache using the `localSchemaCache` instead of `systemCache`? Same as per https://github.com/directus/directus/pull/19480/files#r1689943849 🤔

Reply:
done ✅